### PR TITLE
Add rui-sidenav and allow enter on rui-card [Closes #4 Fixes #23]

### LINKED ISSUE #4: Create Sidenav (rui-sidenav)
* Directive
* Options
* HTML
* CSS

### LINKED ISSUE #23: rui-card should submit on pressing enter

### PR DESCRIPTION
**Pull request by jdl031**

### CODE REVIEW

**Comment by jdl031** [comment]
There's an issue with using the same state to handle both the actual state link as well as the active class when the state reference needs to take parameters. Will fix tomorrow

File: src/js/directives/sidenav.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+var app = angular.module('ruiComponents');
+
+
+app.directive('ruiSidenav', ['$compile', function ($compile) {
+	return {
+		restrict: 'AE',
+		scope: {
+      data: '='
+    },
```
Comment (ajkitson) [inline src/js/directives/sidenav.js]: what's up with the indentation here?

File: src/js/directives/sidenav.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			for (i = 0 ;i < scope.sidenavData.sections.length; i++){
+				if (scope.sidenavData.sections[i].state){
+					scope.sidenavData.sections[i].statename = scope.sidenavData.sections[i].state.toString().split("(")[0];
+				}
+			}
+			console.log(scope)
+			element.append('<div ng-include="\'templates/sidenav.html\'"></div>');
+			$compile(element.contents())(scope);
+    }
+	};
```
Comment (ajkitson) [inline src/js/directives/sidenav.js]: indentation is odd here, too

File: src/js/directives/sidenav.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    },
+    link: function(scope, element, attrs, ctrl, linker){
+
+      scope.sidenavData = scope.data;
+			for (i = 0 ;i < scope.sidenavData.sections.length; i++){
+				if (scope.sidenavData.sections[i].state){
+					scope.sidenavData.sections[i].statename = scope.sidenavData.sections[i].state.toString().split("(")[0];
+				}
+			}
+			console.log(scope)
```
Comment (ajkitson) [inline src/js/directives/sidenav.js]: :arrow_up:

File: src/js/directives/sidenav.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      state: 'dashboard.organization.index({orgId: session.organization.id})',
+      icon: 'ion-settings',
+      hide: '!session.currentUser().admin'
+    },{
+      title: 'FAQ',
+      href: 'http://faq.redoxengine.com',
+      icon: 'ion-help-circled'
+    }
+  ]
+}
```
Comment (ajkitson) [inline src/js/directives/sidenav.js]: :+1: for helpful comments

File: src/js/directives/sidenav.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    },
+    link: function(scope, element, attrs, ctrl, linker){
+
+      scope.sidenavData = scope.data;
+      for (i = 0 ;i < scope.sidenavData.sections.length; i++){
+        if (scope.sidenavData.sections[i].state){
+          scope.sidenavData.sections[i].statename = scope.sidenavData.sections[i].state.toString().split("(")[0];
+        }
+      }
+      console.log(scope)
```
Comment (ajkitson) [inline src/js/directives/sidenav.js]: missed a console.log

File: src/templates/_all.html [hunk 6]
```diff
@@ -182,4 +181,19 @@ <h2 class="page-header">Spinner: <code>rui-spinner</code></h2>
       <rui-spinner ng-show="showInlineSpinner" fullscreen></rui-spinner>
   </div>
 
+  <style media="screen">
+    /* Hack the sidebar into place. */
+    rui-sidenav .rui-sidenav{
+      position: static !important;
+    }
```
Comment (ajkitson) [inline src/templates/_all.html]: this is just to get it so it's not positioned as a sidebar, right?

File: src/templates/sidenav.html [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<div class='rui-sidenav'>
+  <perfect-scrollbar class="sidenav">
+    <div class="breadcrumbs">
+      <a ng-if="sidenavData.breadcrumb.previousState" ui-sref="{{sidenavData.breadcrumb.previousState}}">{{sidenavData.breadcrumb.previousStateTitle | characters:11}}</a>
+			<span ng-if="sidenavData.breadcrumb.previousState" class="ion-android-arrow-forward"></span>
+  		<a ui-sref="{{sidenavData.breadcrumb.state}}">{{sidenavData.breadcrumb.title | characters:11}}</a>
+  	</div>
+    <ul>
```
Comment (ajkitson) [inline src/templates/sidenav.html]: another section where spacing is off :hankey:

File: src/templates/_all.html (hunk 6 above)
Comment (jdl031) [inline src/templates/_all.html]: Correct.

File: src/templates/sidenav.html (hunk 7 above)
Comment (jdl031) [inline src/templates/sidenav.html]: :rage1:

**Comment by ajkitson** [comment]
Looks good. :rocket:

**Pull request merged**